Pearson-Advance/frontend-app-instructors-portal
# refactor: adjustments in events endpoints call

# Description

In this PR is refactored the way in how the service for the events is being used.

This PR resolves [PADV-1666](https://agile-jira.pearson.com/browse/PADV-1666)

## Change log
- Refactor in `Availability` component.
- Refactor in `src/features/Instructor/data/thunks.js` to get all the possible events

### Visual results
No visual changes

### How to test

Clone the repo
```Bash
     git clone <url>
```
Install the packages 📦.
```Bash
     npm i
```
Start project.
```Bash
     npm start
```

Finally, go to [http://localhost:1980/<](http://localhost:1990/my-profile?institutionId=1) and:

- Create multiple events, as many as possible to have multiple pages, you should be able to see all the events in the calendar no matter in which page these events are.
- Do the classic CRUD operations and you should be able to run this scenarios successfully:
    - Get all the events 
    - Create an event ( this should not trigger a get request )
    - Create a recurrent event ( this MUST trigger a get request or multiple get in case to have multiple pages )
    - Edit a single event ( this should not trigger a get request )
    - Edit a recurrent event ( this MUST trigger a get request or multiple get in case to have multiple pages )
    - Delete a single event ( this should not trigger a get request )
    - Delete all events ( this should not trigger a get request )
